tap_github/streams.py

```diff
@@ -412,3 +401,61 @@ def post_process(self, row: dict, context: Optional[dict] = None) -> dict:
             ),
         ),
     ).to_dict()
+
+
+class IssueEventsStream(GitHubStream):
+    """
+    Defines 'IssueEvents' stream.
+    Issue events are fetched from the repository level (as opposed to per issue)
+    to optimize for API quota usage.
+    """
+
+    name = "issue_events"
+    path = "/repos/{org}/{repo}/issues/events"
+    primary_keys = ["id"]
+    replication_key = "created_at"
+    parent_stream_type = RepositoryStream
+    state_partitioning_keys = ["repo", "org"]
+    ignore_parent_replication_key = False
+
+    def get_records(self, context: Optional[dict] = None) -> Iterable[Dict[str, Any]]:
+        """Return a generator of row-type dictionary objects.
+
+        Each row emitted should be a dictionary of property names to their values.
+        """
+        if context and context.get("events", None) == 0:
+            self.logger.debug(f"No events detected. Skipping '{self.name}' sync.")
+            return []
+
+        return super().get_records(context)
+
+    def post_process(self, row: dict, context: Optional[dict] = None) -> dict:
+        row["issue_number"] = int(row["issue"]["number"])
+        row["issue_url"] = row["issue"]["url"]
```

Comment:
Actually @aaronsteers an alternative would be to "duplicate" them and add them to the schema under the "issue" object. What do you think?

Reply:
@ericboucher - Yeah, I'm fine with either approach. We've so far mostly been biasing towards keeping fidelity with the upstream API properties - while also important top-level properties for core keys - and I'm fine to do the same here.

My original suggestion to remove was based on the fact that they were not noted in the schema (to prevent the warning of dropping them when undeclared).

Reply:
Yep, that makes sense. I think I'll add them to the schema and "unpop" them then for consistency.

Reply:
@ericboucher 

> How do you usually approach this problem?

There are a few general principles I try to respect, but in this case they are somewhat in conflict:

1. change as little as possible during EL - _except_...
2. keys should always be top-level entities (includes anything used in identify, incremental replication, storage, or partitioning) 
3. we should exclude derived data which may conflict with the same data pulled from its source of truth (another stream perhaps) - unless immutable or a foreign key, or unless needed for historical audit purposes
4. exclude data only if there's a strong reason to exclude

So, yeah, I don't think we should pull in the full issues record here since there's a full stream dedicated to that data. There are other places in this API where we could also trim back redundant data points but I haven't felt strongly enough that I would slow things down or insist strongly to remove something. (Especially since I'm not intimately familiar with many of the conventions of the API.)